test/server/test-run-test.js

```diff
@@ -0,0 +1,248 @@
+                expect(err).equal(actionError);
+
+                testRun._done();
+                return nextTick();
+            })
+            .then(function () {
+                expect(testRunDone).to.be.true;
+                expect(commandsToClient.length).eql(4);
+                expect(commandsToClient[0]).eql(click1);
+                expect(commandsToClient[1]).eql(click1);
+                expect(commandsToClient[2]).eql(click2);
+                expect(commandsToClient[3].type).eql(commands.TYPE.testDone);
+            });
+    });
+
+    it('Should call test.fn() and complete the test after it is resolved', function () {
+        var testMock = {
+            fixture: 'fixture',
+            fn:      Promise.resolve
+        };
+
+        var testRun     = new TestRun(testMock);
+        var testRunDone = false;
+
+        testRun.once('done', function () {
+            testRunDone = true;
+        });
+
+        return testRun[CLIENT_MESSAGES.ready]()
+            .then(function (command) {
+                expect(command.type).eql(commands.TYPE.testDone);
+                expect(testRunDone).to.be.true;
+            });
+    });
+
+    it('Should call test.fn() and complete the test with an error after it is rejected', function () {
+        var testError = 'test-error';
+        var testMock  = {
+            fixture: 'fixture',
+
+            fn: function () {
+                return Promise.reject(testError);
+            }
+        };
+
+        var testRun     = new TestRun(testMock);
+        var testRunDone = false;
+
+        testRun.once('done', function () {
+            testRunDone = true;
+        });
+
+        return testRun[CLIENT_MESSAGES.ready]()
+            .then(function (command) {
+                expect(testRun.errs[0]).eql(testError);
+                expect(command.type).eql(commands.TYPE.testDone);
+                expect(testRunDone).to.be.true;
+            });
+    });
+
+    it('Should execute the wait command without sending request to the client', function () {
+        var waitingStartedTime = Date.now();
+
+        var testMock = {
+            fixture: 'fixture',
+            fn:      function (testRun) {
+                return testRun
+                    .executeCommand(commands.createWaitCommand(50))
+                    .then(function () {
+                        expect(Date.now() - waitingStartedTime).to.be.least(50);
+                        testRun._done();
+                    });
+            }
+        };
+
+        var testRun = new TestRun(testMock);
+
+        return testRun[CLIENT_MESSAGES.ready]()
+            .then(function (command) {
+                expect(command.type).eql(commands.TYPE.testDone);
+                expect(Date.now() - waitingStartedTime).to.be.least(50);
+                expect(testRun.errs.length).eql(0);
+            });
+    });
+
+    it('Should raise an error when try to execute a new command while the previous on is not finished', function () {
```

Comment:
Should raise an error on an attempt to execute a new command while the previous command execution is not yet finished.